### DISCUSSION TITLE
Stabilozine does almost nothing to stop/slow Cyanide poisoning

### DISCUSSION THREAD

**Discussion by ubertpendragon** [FakeFishGames/Barotrauma#14250] [Ideas and suggestions]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

Stabilozine does -1 strength per second to all poison afflictions, buy Cyanide poisoning has values in the +2/3/4/5 per second,
It significantly overpowers stabilozine to the point stabilozine does nothing against it. It also means its antidote (at -10/s) is worse than other antidotes.

**You have very little time to react with stabilozine, and even if you do the cyanide kills you anyway**

Without stabilozine, Cyanide kills in ~27s
With stabilozine, Cyanide kills in ~41s (IF APPLIED BEFORE CYANIDE)
It takes about 16s for cyanide to react 50, where it starts becoming overwhelmingly lethal due to organ damage

Values should be rescaled so that cyanide retains a fast TTK, but stabilozine be actually useful against it, either stopping cyanide progression or slowing it down significantly.

From:
```
0 - 10: -0.25/s
10 - 30: 2/s
30 - 50: 3/s
50 - 75: 4/s
75 - 100: 5/s
```

To:
```
0 - 10: -0.25/s
10 - 20: 1/s
20 - 30: 1/s  
40- 50: 1/s   (organ damage 30 => 5 otherwise you just die, stabilozine or not)
50- 100: 1/s (organ damage overpowers everything here)

Starts organ damage at 20s
36s of TTK (still by far THE fastest poison)

You could also reduce  showiconthreshold="15" so cyanide poisoning shows up earlier and leaves time to react
```

```xml
    <Effect minstrength="0" maxstrength="10"
      strengthchange="-0.25"
      minscreendistort="0.0"
      maxscreendistort="0.4"
      minscreenblur="0.0"
      maxscreenblur="0.4"
      minfacetint="0,100,180,0"
      maxfacetint="0,100,180,20">
      <StatusEffect target="Character" multiplyafflictionsbymaxvitality="true">
        <Affliction identifier="oxygenlow" amount="11" />
      </StatusEffect>
    </Effect>
    <Effect minstrength="10" maxstrength="20"
      strengthchange="1"
      minscreendistort="0.4"
      maxscreendistort="0.0"
      minscreenblur="0.4"
      maxscreenblur="0.0"
      minfacetint="0,100,180,20"
      maxfacetint="0,100,180,40"
      minbodytint="0,100,180,0"
      maxbodytint="0,100,180,20"
      minspeedmultiplier="1.0"
      maxspeedmultiplier="0.9">
      <StatusEffect target="Character" multiplyafflictionsbymaxvitality="true">
        <Affliction identifier="oxygenlow" amount="11" />
      </StatusEffect>
      <StatusEffect target="Character" interval="1" disabledeltatime="true">
        <ReduceAffliction identifier="huskinfection" amount="4" />
      </StatusEffect>
    </Effect>
    <Effect minstrength="20" maxstrength="30"
      strengthchange="1"
      minscreendistort="0.0"
      maxscreendistort="0.8"
      minscreenblur="0.0"
      maxscreenblur="0.8"
      minfacetint="0,100,180,40"
      maxfacetint="0,100,180,60"
      minbodytint="0,100,180,0"
      maxbodytint="0,100,180,40"
      minspeedmultiplier="0.9"
      maxspeedmultiplier="0.7">
      <StatusEffect target="Character" multiplyafflictionsbymaxvitality="true">
        <Affliction identifier="oxygenlow" amount="10" />
      </StatusEffect>
      <StatusEffect target="Character" interval="1" disabledeltatime="true">
        <ReduceAffliction identifier="huskinfection" amount="8" />
      </StatusEffect>
    </Effect>
    <Effect minstrength="30" maxstrength="40"
      strengthchange="1"
      minscreendistort="0.8"
      maxscreendistort="1.0"
      minscreenblur="0.8"
      maxscreenblur="1.0"
      minfacetint="0,100,180,60"
      maxfacetint="0,100,180,100"
      minbodytint="0,100,180,40"
      maxbodytint="0,100,180,80"
      minspeedmultiplier="0.7"
      maxspeedmultiplier="0.7"
      tag="poisoned">
      <StatusEffect target="Character" multiplyafflictionsbymaxvitality="true">
        <Affliction identifier="oxygenlow" amount="10" />
      </StatusEffect>
      <StatusEffect target="Character" setvalue="true">
        <Conditional healthpercentage="lt 75"/>
        <Affliction identifier="stun" amount="1" />
      </StatusEffect>
      <StatusEffect target="Character">
        <Affliction identifier="organdamage" amount="5" />
      </StatusEffect>
      <StatusEffect target="Character" interval="1" disabledeltatime="true">
        <ReduceAffliction identifier="huskinfection" amount="20" />
      </StatusEffect>
    </Effect>
    <Effect minstrength="40" maxstrength="50"
      strengthchange="1"
      minscreendistort="1.0"
      maxscreendistort="1.0"
      minscreenblur="1.0"
      maxscreenblur="1.0"
      minfacetint="0,100,180,100"
      maxfacetint="0,100,180,100"
      minbodytint="0,100,180,80"
      maxbodytint="0,100,180,80"
      minspeedmultiplier="0.7"
      maxspeedmultiplier="0.7"
      tag="poisoned">
      <StatusEffect target="Character" multiplyafflictionsbymaxvitality="true">
        <Affliction identifier="oxygenlow" amount="10" />
      </StatusEffect>
      <StatusEffect target="Character" setvalue="true">
        <Conditional healthpercentage="lt 90"/>
        <Affliction identifier="stun" amount="1" />
      </StatusEffect>
      <StatusEffect target="Character">
        <Affliction identifier="organdamage" amount="10" />
      </StatusEffect>
      <StatusEffect target="Character" interval="1" disabledeltatime="true">
        <ReduceAffliction identifier="huskinfection" amount="50" />
      </StatusEffect>
    </Effect>
    <Effect minstrength="50" maxstrength="100"
      strengthchange="1"
      minscreendistort="1.0"
      maxscreendistort="1.0"
      minscreenblur="1.0"
      maxscreenblur="1.0"
      minfacetint="0,100,180,100"
      maxfacetint="0,100,180,100"
      minbodytint="0,100,180,80"
      maxbodytint="0,100,180,80"
      minspeedmultiplier="0.7"
      maxspeedmultiplier="0.7"
      tag="poisoned">
      <StatusEffect target="Character" multiplyafflictionsbymaxvitality="true">
        <Affliction identifier="oxygenlow" amount="10" />
      </StatusEffect>
      <StatusEffect target="Character" setvalue="true">
        <Conditional healthpercentage="lt 90"/>
        <Affliction identifier="stun" amount="1" />
      </StatusEffect>
      <StatusEffect target="Character">
        <Affliction identifier="organdamage" amount="60" />
      </StatusEffect>
      <StatusEffect target="Character" interval="1" disabledeltatime="true" multiplyafflictionsbymaxvitality="true" ConditionalComparison="And">
        <Conditional IsHuman="false"/>
        <Conditional healthpercentage="gt 50"/>
        <Affliction identifier="cyanidepoisoning" amount="-9999" probability="0.075" />
      </StatusEffect>
      <StatusEffect target="Character" interval="1" disabledeltatime="true">
        <ReduceAffliction identifier="huskinfection" amount="100" />
      </StatusEffect>
    </Effect>
```

### Reproduction steps

Inject cyanide
Try to stall it with stabilozine
Just die

### Version

v1.5.9.1 (Summer Update Hotfix 2)

**Comment by Regalis11** [comment]
I'm not sure I see why you consider this a bug. The description of stabilozine says "can be used to slow down the effects of most toxins" - it doesn't say it stops or even effectively slows down all types of poisonings. I think it's actually a good thing there are poisons with different effects, and that not all meds affect all of them in the same way.

**Reply by ubertpendragon** [reply]
There's a world between "slowing down" and doing nothing, stabilozine does *nothing* to cyanide

Slowing down = cyanide growth at 1.1 per second
Doing nothing =cyanide growth at 3/4/5 per second

Stabilozine is still listed as a potential treatment for cyanide, the wiki itself is also confusing in that regard. 

> I think it's actually a good thing there are poisons with different effects, and that not all meds affect all of them in the same way.

Honestly it just makes cyanide THE poison. Kills super fast, impossible to stop, slow down or cure, and being "expensive" also means the antidote itself is expensive.

**Reply by Regalis11** [reply]
> Without stabilozine, Cyanide kills in ~27s
> With stabilozine, Cyanide kills in ~41s (IF APPLIED BEFORE CYANIDE)

To me this does not seem like "nothing", even though you're right that it such a minor effect doesn't do much in practice.

Cyanide being "the poison" is a good point though, and something we probably need to do something about. I don't think it's necessarily a bad that it's hard to cure and kills fast, nor that stabilozine isn't very useful to slow it down, but those should probably be compensated by something else to prevent it from being the clearly most effective poison.

**Reply by SomeRandomNoobKekeke** [reply]
It's maybe the fastest to kill and the hardest to cure, but 
It's not the sneaky kill without -rep poison
it's not the low med skill friendly poison
It's not the cheapest poison
So it's not the poison

**Reply by ubertpendragon** [reply]
> To me this does not seem like "nothing", even though you're right that it such a minor effect doesn't do much in practice.

I don't want to get too much into math but:

- It's only detected at 15 in medical scanner or 30 as symptoms, so like 10 seconds in the affliction, which leaves you 17 seconds to react, 

- Every other poison roughly increases at +1/s, and the antidote at -10/s, which means antidote reduces the afflictions by 9/s. Cyanide grows by 1/2/3/4/5. It only shows up when it's acting at 3/s, so the antidote reduces the affliction by -7/s to -5/s, at which point cyanide kills you extremely fast, which makes it way less effective than other antidotes.

It just means that stabilozine as a counter is nonexistent, and the antidote is inefficient, which is what I tried to fix

**Reply by SomeRandomNoobKekeke** [reply]
it literally says:
![Screenshot_4](https://github.com/FakeFishGames/Barotrauma/assets/122838333/e893b819-4ea4-4b12-a346-456c2e7de753)

**Reply by ubertpendragon** [reply]
To simplify even more, Cyanide poisoning effects can't be cured by antidote at around 75% which is 23 seconds

The tooltip appears after 10 seconds
You have around 8 seconds to inject *yourself* with antidote (before you get stunned)
Someone else has 13 seconds to inject the antidote


I changed it so that cyanide doesnt entirely bypass its counters while still being the strongest poison

**Reply by Zimon3283** [reply]
Also viperlings do cyanide and if your hit a few times you can't slow it down enough with stabloizine to
1 kill the viperlings
2 retrieve theyr loot
3make an antidote
4 cure you
So them "luckily dropping the ingredients to the cure" is not really that usefull
